packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageManager.java

```diff
@@ -40,9 +44,34 @@ public void setSource(ImageView view, @Nullable ReadableMap sourceMap) {
       return;
     }
 
+    RequestOptions options = new RequestOptions();
+
+    // Override the size for local assets. This ensures that
+    // resizeMode "center" displays the image in the correct size.
+    if (sourceMap.hasKey(SOURCE_WIDTH_KEY) && sourceMap.hasKey(SOURCE_HEIGHT_KEY) && sourceMap.hasKey(SOURCE_SCALE_KEY)) {
+      double scale = sourceMap.getDouble(SOURCE_SCALE_KEY);
+      int width = sourceMap.getInt(SOURCE_WIDTH_KEY);
+      int height = sourceMap.getInt(SOURCE_HEIGHT_KEY);
+      options.override((int) (width * scale), (int) (height * scale));
+    }
+
+    options.fitCenter();
+
     mRequestManager
-        .load(sourceMap.getString(SOURCE_URI_KEY))
-        .into(view);
+      .load(sourceMap.getString(SOURCE_URI_KEY))
+      .apply(options)
+      .into(view);
+  }
+
+  @ReactProp(name = "resizeMode")
+  public void setResizeMode(ImageView view, String resizeMode) {
+    ImageView.ScaleType scaleType;
+    if (resizeMode.equals("repeat")) {
+      scaleType = ImageView.ScaleType.FIT_XY;
+    } else {
+      scaleType = ExpoImageTypes.getScaleType(resizeMode);
+    }
+    view.setScaleType(scaleType);
```

Comment:
This implementation differs from the Android one when it comes to when the prop changes are applied. We could probably let it go, if this PR lands before the events PR I will simply rebase and split these changes.

Reply:
That is correct. There will be a lot more differences between the iOS/Android implementations at the view/drawing level